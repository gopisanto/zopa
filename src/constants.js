export const COUNTRY_CURRENCY_CODE = 'en-UK';
export const CURRENCY_FORMAT_OPTIONS = {
  symbol: 'Â£',
  style: 'currency',
  currency: 'GBP',
  minimumFractionDigits: 4,
  maximumFractionDigits: 4,
};
