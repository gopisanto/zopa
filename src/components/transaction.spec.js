import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Transaction from './transaction';
import { COUNTRY_CURRENCY_CODE, CURRENCY_FORMAT_OPTIONS } from '../constants';

Enzyme.configure({ adapter: new Adapter() });

let transaction;

describe('Transaction component', () => {
  beforeEach(() => {
    transaction = {
      name: 'ZOPA',
      email: 'zopa@zopa.de',
      amountTransferred: 5000,
    }
  });

  test('should render', () => {
    const wrapper = shallow(
      <Transaction transaction={transaction} />
    );

    expect(wrapper.prop('className')).toEqual('transaction');
  });

  test('should render same snapshot', () => {
    const wrapper = renderer.create(
      <Transaction transaction={transaction} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should render transaction details', () => {
    const wrapper = shallow(
      <Transaction transaction={transaction} />
    );

    expect(wrapper.find('.name').text()).toEqual(transaction.name);
    expect(wrapper.find('.email').text()).toEqual(transaction.email);
    expect(wrapper.find('.amount').text()).toEqual(
      transaction.amountTransferred.toLocaleString(COUNTRY_CURRENCY_CODE, CURRENCY_FORMAT_OPTIONS)
    );
  });
});
