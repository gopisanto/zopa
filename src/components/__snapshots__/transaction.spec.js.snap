// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should render same snapshot 1`] = `
<div
  className="transaction"
>
  <div
    className="person"
  >
    <label
      className="name"
    >
      ZOPA
    </label>
    <label
      className="email"
    >
      zopa@zopa.de
    </label>
  </div>
  <div
    className="amount"
  >
    <label>
      Â£5,000.0000
    </label>
  </div>
</div>
`;
